server/data/query.ts

```diff
@@ -55,6 +58,31 @@ export async function getQuery(displayId: string): Promise<Query> {
   throw new Error(`${displayId} not found.`);
 }
 
+export async function getQueryByUUID(uuid: string): Promise<Query> {
+  const supabase = await buildSupabaseServerClient();
+
+  const { status, data, error } = await supabase
+    .from("queries")
+    .select("*")
+    .eq("uuid", uuid)
+    .limit(1)
+    .maybeSingle();
+
+  if (error) {
+    console.error(error);
+    throw new Error(
+      `Error fetching query with UUID= ${uuid}: ${error.message}`,
+    );
+  } else if (status === 200) {
+    if (data) {
+      return processQueryData(data);
+    } else {
+      notFound();
```

Comment:
that's a little weird but I guess the status of a sql query that returns no data does not have to be 404

Reply:
this actually does return a 404: https://nextjs.org/docs/app/api-reference/file-conventions/not-found

Reply:
then it is probably better to check the status rather than request body before showing `not found`?

Reply:
The query doesn't throw an error if nothing is found, so you check to see if it's empty. We could have it throw an error, by calling `single()` instead of `maybeSingle()`, but then we have to start parsing the error details to distinguish between not found and any other database error.